users/serializers.py

```diff
@@ -10,3 +22,72 @@ def get_token(cls: Any, user: Any) -> Any:
         token["userID"] = user.id
         token["editor"] = user.is_editor
         return token
+
+
+class PasswordResetRequestSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=True)
+
+    class Meta:
+        fields = ["email"]
+
+    def validate(self, attrs: Any) -> Any:
+        try:
+            request = self.context.get("request")
+            email = attrs.get("email")
+            if User.objects.filter(email=email).exists():
+                user = User.objects.get(email=email)
+                uidb64 = urlsafe_base64_encode(smart_bytes(user.id))
+                token = PasswordResetTokenGenerator().make_token(user)
```

Comment:
You can have some of this functionality in a staticmethod so that the validate method is not so big so that is easier to follow along